File: ios/Infrastructure/Models/MediaType/MediaType.swift
//
//  MediaType.swift
//  KlipyiOS-demo
//
//  Created by Tornike Gomareli on 13.01.25.
//


import Foundation

public enum MediaType: String, Codable {
  case clips = "clip"
  case gifs = "gif"
  case stickers = "sticker"
  case ad = "ad"
  
  var path: String {
    switch self {
    case .clips: return "clips"
    case .gifs: return "gifs"
    case .stickers: return "stickers"
    case .ad: return ""
    }
  }
}

extension MediaType {
  var displayName: String {
    switch self {
    case .clips: return "CLIPs"
    case .gifs: return "GIFs"
    case .stickers: return "STICKERs"
    case .ad:
      return ""
    }
  }
  
  static func from(string: String) -> MediaType? {
    return MediaType(rawValue: string.lowercased())
  }
}

File: android/src/main/java/com/klipy/presentation/features/conversation/model/MediaType.kt
package com.klipy.presentation.features.conversation.model

enum class MediaType(val title: String) {
    GIF("GIFs"), CLIP("Clips"), STICKER("Stickers"), AD("Ads")
}

fun MediaType.getSingularName(): String {
    return when (this) {
        MediaType.GIF -> "GIF"
        MediaType.STICKER -> "Sticker"
        MediaType.CLIP -> "Clip"
        MediaType.AD -> "AD"
    }
}